build/Targets/Settings.props

```diff
@@ -13,8 +13,10 @@
     <NuGetToolPath>$(RepoRoot)nuget.exe</NuGetToolPath>
     <ToolsetPackagesDir>$(RepoRoot)build\ToolsetPackages\</ToolsetPackagesDir>
     <ToolsetPackagesSemaphore>$(ToolsetPackagesDir)toolsetpackages.semaphore</ToolsetPackagesSemaphore>
-    <ToolsetCompilerPackageName>Microsoft.Net.Compilers</ToolsetCompilerPackageName>
-    <ToolsetCompilerPackageVersion>2.0.1</ToolsetCompilerPackageVersion>
+    <ToolsetCompilerPackageName Condition="'$(OS)' == 'Windows_NT'">Microsoft.Net.Compilers</ToolsetCompilerPackageName>
+    <ToolsetCompilerPackageName Condition="'$(OS)' != 'Windows_NT'">Microsoft.NETCore.Compilers</ToolsetCompilerPackageName>
+    <ToolsetCompilerPackageVersion Condition="'$(OS)' == 'Windows_NT'">2.0.1</ToolsetCompilerPackageVersion>
+    <ToolsetCompilerPackageVersion Condition="'$(OS)' != 'Windows_NT'">2.3.0-dev-56735-00</ToolsetCompilerPackageVersion>
```

Comment:
Doesn't look like these properties are used here.  Can we group them together with the `$(ToolsetCmpilerPropsFilePath)` property on line 40?